File: ruby/pe006.rb
#!/usr/bin/env ruby
# http://projecteuler.net/problem=6
limit = 100
sum_of_squares = (1..limit).inject { |sum, n| sum + n ** 2 }
square_of_sum = (1..limit).inject { |sum, n| sum + n } ** 2
difference = (sum_of_squares - square_of_sum).abs
puts difference

File: go/pe006.go
// http://projecteuler.net/problem=6
package main

import "fmt"

func main() {
	sum_of_squares := 0
	square_of_sums := 0
	for n := 1; n <= 100; n++ {
		sum_of_squares += n * n
		square_of_sums += n
	}
	square_of_sums *= square_of_sums
	fmt.Println(square_of_sums - sum_of_squares)
}
